gen/build_deploy/bash.py

```diff
@@ -669,24 +668,31 @@ def fill_template(base_name, format_args):
     return installer_filename
 
 
-def make_dcos_launch():
-    # NOTE: this needs to be kept in sync with build_dcos_launch.sh
+def make_dcos_launch(variant, launch_info):
+    util_bootstrap_id = launch_info['bootstrap']
+    launch_package_id = None
+    for package_id in launch_info['packages']:
+        if pkgpanda.PackageId(package_id).name == 'dcos-launch':
+            launch_package_id = package_id
+            break
+    assert launch_package_id, 'dcos-launch package cannot be found!'
+    util_bootstrap_filename = util_bootstrap_id + '.bootstrap.tar.xz'
+    # Make this its own function for easy mocking
     work_dir = py.path.local.mkdtemp()
-    # the tree will not be in the working dir if run from downstream
-    tree_src = './'
-    if os.path.exists('ext/upstream'):
-        tree_src = './ext/upstream'
-    src_info = {'kind': 'git_local', 'rel_path': tree_src}
-    this_tree = GitLocalSrcFetcher(src_info, None, os.getcwd())
-    this_tree.checkout_to(str(work_dir))
-    # put the launch spec into the root of the tree before running pyinstaller
-    work_dir.join('dcos-launch.spec').write(pkg_resources.resource_string('launch', 'dcos-launch.spec'))
+    subprocess.check_call([
+        'cp', os.getcwd() + '/packages/cache/bootstrap/' + util_bootstrap_filename,
+        str(work_dir.join(util_bootstrap_filename))])
     with work_dir.as_cwd():
-        subprocess.check_call(['pyinstaller', '--log-level=DEBUG', 'dcos-launch.spec'])
-    subprocess.check_call(['mv', str(work_dir.join('dist').join('dcos-launch')), 'dcos-launch'])
-    work_dir.remove()
-
-    return "dcos-launch"
+        subprocess.check_call(['xz', '-d', util_bootstrap_filename])
+        subprocess.check_call(['tar', 'xf', util_bootstrap_filename.rstrip('.xz')])
```

Comment:
It is not at all clear where I can find the uncompressed packages and their artifacts from builds. The packages were the only thing being passed around here. If you want to instruct me on how exactly I can go from `variant_arguments` and `all_completes` to having uncompressed packages that were guaranteed to be from this same build, then I would be glad to make the change. Otherwise, this is the best I can do from guessing the magic state of the system my looking at these few lines of code https://github.com/dcos/dcos/blob/master/gen/build_deploy/bash.py#L635 (which don't tell me anything about uncompressed packages).